src/expression.d

```diff
@@ -13976,7 +13985,10 @@ public:
 
     override Expression toBoolean(Scope* sc)
     {
-        e2 = e2.toBoolean(sc);
+        auto ex2 = e2.toBoolean(sc);
```

Comment:
The whole scheme of error reporting in DMD is to mark the trees as TOKerror, and not try and "repair" the trees, ever.